evap/results/views.py

```diff
@@ -95,26 +94,24 @@ def course_detail(request, semester_id, course_id):
         questionnaire_result.question_results = filtered_question_results
 
     # remove empty questionnaire_results
-    questionnaire_results = [questionnaire_result for questionnaire_result in questionnaire_results if questionnaire_result.question_results]
+    for contribution_result in course_result.contribution_results:
+        contribution_result.questionnaire_results = [q for q in contribution_result.questionnaire_results if q.question_results]
```

Comment:
here you're introducing single letter variable names to save space - is this the only case where we're using that? maybe we should stick to the verbose names although the lines are getting longer. if not - this should be at least `qr`.